### DISCUSSION TITLE
Using `docker-outside-of-docker` with rootless docker

### DISCUSSION THREAD

**Discussion by aucampia** [devcontainers/community#52] [Q&A]
Is there any way that `docker-outside-of-docker` could be made to work with rootless docker? What would be needed is a way to specify the `docker.sock` path [[ref](https://github.com/devcontainers/features/blob/b48d8700ffe5c0b363c9e8706ba2c1601a2bb94f/src/docker-outside-of-docker/devcontainer-feature.json#L47-L52)] but I can't see how this can easily be configurable. In other places I use `DOCKER_SOCK` as an environment variable to specify it, if there was some way it could read the value from there it would be great.

**Comment by a-gn** [comment]
Could there be a [pre-defined variable](https://containers.dev/implementors/json_reference/#variables-in-devcontainerjson) for the Docker socket used by VSCode to start the dev container? This would enable something like:

```json
{
    "mounts": [
        {
            "source": "${hostDockerSocket}",
            "target": "/var/run/docker.sock",
            "type": "bind"
        }
    ]
}
```
